test/gtest/sequences.cpp

```diff
@@ -0,0 +1,207 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <miopen/sequences.hpp>
+#include <gtest/gtest.h>
+
+namespace Seq = miopen::seq;
+
+TEST(CPU_sequence_NONE, SimpleListTest)
```

Comment:
Suggest, improves brevity and maintainability (also applies to most other tests):
```
#define SimpleListTest_VALUES 1, 2, -1, 4, 5
TEST(CPU_sequence_NONE, SimpleListTest)
{
    Seq::Sequence<int, SimpleListTest_VALUES> test_seq;
    auto expected = std::vector<int>{SimpleListTest_VALUES};

    auto it = test_seq.begin();
    for(auto v : expected)  EXPECT_EQ(*(it++), v);
    EXPECT_EQ(it++, test_seq.end());  // may actually want explicit tests that define and enforce behavior of traveling past the ends of the sequence
}
```
We should also talk about other possibilities, for example:
- preincrement vs. postincrement
- decrement (probably not intended to function, but it does)
- other behaviors in comparison with STL containers